services/wallet/activity/activity_v2.go

```diff
@@ -72,14 +76,89 @@ func getActivityEntriesV2(ctx context.Context, deps FilterDependencies, addresse
 
 	qConditions = append(qConditions, sq.Eq{"rpt.chain_id": chainIDs})
 	qConditions = append(qConditions, sq.Eq{"rip.from_address": addresses})
+	// Only fetch entries after the specified timestamp
+	qConditions = append(qConditions, sq.Gt{"tt.timestamp": afterTimestamp})
```

Comment:
hmmm I see you're REALLY trying to avoid merging the results :D

I think this approach will fail if you get some fetched tx while a previous sent tx is still pending (some chains are much slower to confirm transactions than others). You can get some weird behavior like a pending Tx disappearing from the list and only appearing once it's confirmed and part of the fetched list.

Reply:
Yes, you really can blame on this :) From my pov the merging of database entities is much harder to comprehend when trying to understand and fix the code, especially if you haven't written it. I agree that performance would be somewhat worse but I think that change is not that serious because we are talking about tens, maybe hundreds of entities processed once in 10 minutes. And making performance optimisations here before we see that it makes app slower I would consider as "preliminary". I'm pretty sure the app has much more sever bottlenecks.

Thanks for pointing, yes, this could be a bug for sure. I will add some tests to cover such scenario and check that merging works properly.